cmd/tempo/app/overrides_validation.go

```diff
@@ -33,9 +33,9 @@ func (r *runtimeConfigValidator) Validate(config *overrides.Overrides) error {
 		}
 	}
 
-	if config.MetricsGenerator.GenerateNativeHistograms != "classic" &&
-		config.MetricsGenerator.GenerateNativeHistograms != "native" &&
-		config.MetricsGenerator.GenerateNativeHistograms != "both" &&
+	if config.MetricsGenerator.GenerateNativeHistograms != overrides.HistogramMethodClassic &&
```

Comment:
Not sure if this is better but... we could also use the `overrides.HistogramModeToValue` to check if the provided value is a valid key. By checking whether the map contains the given value as a key.
That would mean if we update the map and add a key (which is fairly unlikely tbh) this validation will automatically be updated.

Reply:
This should work.
```golang
	if v, ok := registry.HistogramModeToValue[string(config.MetricsGenerator.GenerateNativeHistograms)]; !ok {
		if string(v) != "" {
			return fmt.Errorf("metrics_generator.generate_native_histograms \"%s\" is not a valid value, valid values: classic, native, both", config.MetricsGenerator.GenerateNativeHistograms)
		}
	}
```
Do you like that better?

Reply:
Yep that should work :+1: Readability suffers a bit but that's more of a Go issue due to its limited enum support.